hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/lock/OzoneManagerLock.java

```diff
@@ -195,18 +202,28 @@ private boolean lock(Resource resource, String resourceName,
     }
   }
 
-  private void updateReadLockMetrics(Resource resource, String lockType,
+  private void updateReadLockMetrics(Resource resource,
                                      long startWaitingTimeNanos) {
-    if (lockType.equals(READ_LOCK)) {
-      long readLockWaitingTimeNanos =
-          Time.monotonicNowNanos() - startWaitingTimeNanos;
+    long readLockWaitingTimeNanos =
+        Time.monotonicNowNanos() - startWaitingTimeNanos;
 
-      // Adds a snapshot to the metric readLockWaitingTimeMsStat.
-      omLockMetrics.setReadLockWaitingTimeMsStat(
-          TimeUnit.NANOSECONDS.toMillis(readLockWaitingTimeNanos));
+    // Adds a snapshot to the metric readLockWaitingTimeMsStat.
+    omLockMetrics.setReadLockWaitingTimeMsStat(
+        TimeUnit.NANOSECONDS.toMillis(readLockWaitingTimeNanos));
 
-      resource.setStartHeldTimeNanos(Time.monotonicNowNanos());
-    }
+    resource.setStartHeldTimeNanos(Time.monotonicNowNanos());
+  }
+
+  private void updateWriteLockMetrics(Resource resource,
+                                      long startWaitingTimeNanos) {
+    long writeLockWaitingTimeNanos =
+        Time.monotonicNowNanos() - startWaitingTimeNanos;
+
+    // Adds a snapshot to the metric writeLockWaitingTimeMsStat.
+    omLockMetrics.setWriteLockWaitingTimeMsStat(
+        TimeUnit.NANOSECONDS.toMillis(writeLockWaitingTimeNanos));
+
+    resource.setStartHeldTimeNanos(Time.monotonicNowNanos());
```

Comment:
`resource.setStartHeldTimeNanos(Time.monotonicNowNanos());`

```
OzoneManagerLock.java

    void setStartHeldTimeNanos(long startHeldTimeNanos) {
      readLockTimeStampNanos.get().setStartHeldTimeNanos(startHeldTimeNanos);
    }
```
Its overwriting the held time for the read metrics. Please create a separate var `writeLockTimeStampNanos`. Since we have plan to allow multiple writers eventually(long term gaol) on disjoint OBS/FSO paths, we need to use `ThreadLocal<LockUsageInfo> writeLockTimeStampNanos`, so that we keep track of multiple writer threads stats.

Reply:
Thank you for the comment, I have incorporated the changes!